src/lib/API/api.js

```diff
@@ -530,6 +533,10 @@ export class APIService {
         baseURL: explorer,
       })
 
+      if (!isArray(events)) {
+        return
+      }
+
```

Comment:
1. added it to getAbi > used by walletconnectclient which already has a catch flow and log.error
2. getsourcecode seems to not need it since it returns res?.ContractName so would just result in undefined. (do we still need an error handler here?)
3. altered the safeguard for getLogs to only add the events which were successfully fetched

Reply:
@L03TJ3 you are not addressing my points
1. decision between returning null, throwing exception or events collected until error based on how caller uses results, retry requirements or lastBlock
2. we have multiple places where we call APIs, tatum, celo, fuse